Dalamud/Interface/Internal/Windows/Data/DataWindow.cs

```diff
@@ -0,0 +1,194 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+using Dalamud.Game.Gui;
+using Dalamud.Interface.Components;
+using Dalamud.Interface.Windowing;
+using Dalamud.Utility;
+using ImGuiNET;
+using Serilog;
+
+namespace Dalamud.Interface.Internal.Windows;
+
+/// <summary>
+/// Class responsible for drawing the data/debug window.
+/// </summary>
+internal class DataWindow : Window
+{
+    private readonly IDataWindowWidget[] modules =
+    {
+        new ServerOpcodeWidget(),
+        new AddressesWidget(),
+        new ObjectTableWidget(),
+        new FateTableWidget(),
+        new SeFontTestWidget(),
+        new FontAwesomeTestWidget(),
+        new PartyListWidget(),
+        new BuddyListWidget(),
+        new PluginIpcWidget(),
+        new ConditionWidget(),
+        new GaugeWidget(),
+        new CommandWidget(),
+        new AddonWidget(),
+        new AddonInspectorWidget(),
+        new AtkArrayDataBrowserWidget(),
+        new StartInfoWidget(),
+        new TargetWidget(),
+        new ToastWidget(),
+        new FlyTextWidget(),
+        new ImGuiWidget(),
+        new TexWidget(),
+        new KeyStateWidget(),
+        new GamepadWidget(),
+        new ConfigurationWidget(),
+        new TaskSchedulerWidget(),
+        new HookWidget(),
+        new AetherytesWidget(),
+        new DtrBarWidget(),
+        new UIColorWidget(),
+        new DataShareWidget(),
+    };
+
+    private readonly Dictionary<DataKind, string> dataKindNames = new();
+
+    private bool isExcept;
+    private DataKind currentKind;
+    
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DataWindow"/> class.
+    /// </summary>
+    public DataWindow()
+        : base("Dalamud Data")
+    {
+        this.Size = new Vector2(500, 500);
+        this.SizeCondition = ImGuiCond.FirstUseEver;
+
+        this.RespectCloseHotkey = false;
+
+        foreach (var dataKind in Enum.GetValues<DataKind>())
+        {
+            this.dataKindNames[dataKind] = dataKind.ToString().Replace("_", " ");
+        }
+
+        this.Load();
+    }
+
+    /// <inheritdoc/>
+    public override void OnOpen()
+    {
+    }
+
+    /// <inheritdoc/>
+    public override void OnClose()
+    {
+    }
+
+    /// <summary>
+    /// Set the DataKind dropdown menu.
+    /// </summary>
+    /// <param name="dataKind">Data kind name, can be lower and/or without spaces.</param>
+    public void SetDataKind(string dataKind)
+    {
+        if (string.IsNullOrEmpty(dataKind))
+            return;
+
+        dataKind = dataKind switch
+        {
+            "ai" => "Addon Inspector",
+            "at" => "Object Table", // Actor Table
```

Comment:
this seems a little bit fragile; can we match to the appropriate enum variants here instead?

or, thinking about my previous comment - maybe we can do some runtime initialization with finding classes in the `Dalamud.Interface.Internal.Windows.Data` namespace with an attribute that contains the window name & any shortcuts? i.e.:
```csharp
[DataKind("DTR Bar", new string[]{"dtr"})]
internal class DtrBarWidget : IDataWindowWidget {
```
would love your thoughts on that idea - I'm not sure if it's entirely sane but it seems like it could be a bit nicer?

Reply:
My thought was to expand the IDataWindowWidget interface to include an alias field instead, and to iterate over the known aliases to find the correct tab to be on.
```cs
string[] aliases = { "ai" };
```

Reply:
fair yeah, what I suggested is probably a bit of overkill for something internal like this.

if you want to make the alias change, I'd be okay with it - I just don't really want to be matching to strings in case we decide to rename  `Object Table` to `Baguette Table` or something in the future 😄

Reply:
Agree command handling is a bit messy but also a non-issue I'd feel. We can deal with this if/when it comes up.

Reply:
This can easily be addressed in the future if needed due to the implementation of the Interface.

I'll resolve this for now.